scripts/build_kernel.sh

```diff
@@ -1,35 +1,47 @@
 #!/usr/bin/env bash
 
+SUPPORTED_FLAVOURS=( v8 2712 )
+FLAVOUR=$1
+
+case " ${SUPPORTED_FLAVOURS[@]} " in
+    (*" $FLAVOUR "*) echo "Building for kernel version: $FLAVOUR" ;;
+    (*)
+        echo "Kernel version not supported: $FLAVOUR"
+        echo "Supported versions: ${SUPPORTED_FLAVOURS[@]}"
+        exit 1
+        ;;
+
+esac
+
 apt-get update
 
-apt-cache depends linux-image-rpi-v8 | grep Depends: > deb.list
+apt-cache depends linux-image-rpi-"$FLAVOUR" | grep Depends: > deb.list
 
 sed -i -e 's/[<>|:]//g' deb.list
 sed -i -e 's/Depends//g' deb.list
 sed -i -e 's/ //g' deb.list
 filename="deb.list"
 
-# here the file will contain something like linux-image-6.6.20+rpt-rpi-v8
+# here the file will contain something like linux-image-6.6.20+rpt-rpi-<version>
 # we want to check if the deb file already exists, eventually build
 while read -r line
 do
     name="$line"
-    if find /output -name "$name*.deb" -printf 1 -quit | grep -q 1
+    if find /output/"$FLAVOUR" -name "$name*.deb" -printf 1 -quit | grep -q 1
     then
-        echo Kernel packages are already available. Nothing to do here.
+        echo "Kernel packages are already available. Nothing to do here."
         exit 0
     fi
 done < "$filename"
 
 # If we get here we didn't find the relevant deb package. Build and install it
 apt-get update
-apt-get source linux-image-rpi-v8
+apt-get source linux-image-rpi-"$FLAVOUR"
 
 # From
 #   https://wiki.debian.org/HowToCrossBuildAnOfficialDebianKernelPackage
 ARCH=arm64
 FEATURESET=rpi
```

Comment:
Can you move to the top also these ENV vars to keep all of them together?

Reply:
Done ✅